### ISSUE TITLE
Normalization method for softmax cross entropy is confusing

### ISSUE THREAD

**Issue by unnonouno** [chainer/chainer#982]
Softmax cross entropy (SCE) calculates average loss of labels. When an ignored label is includes in Variables, it calculates average loss of Variables except ignored one.
For example, `x0 = 0, x1 = -1, x2 = 2` and ignored label is `-1`, its loss is `(loss(x0) + loss(x2)) / 2`.

When we calculates sum of SEC for a sequence, it causes a problem.
When `x0 = 0, x1 = 2, x2 h = 1` and `y0 = 2, y1 = -1, y2 = 0`, `sce(x) + sce(y) = (loss(x0) + loss(x1) + loss(x2)) / 3 + (loss(y0) + loss(y1)) / 2`. So, `y` has larger loss than `x`.
